testing-rspec/README.md

```diff
@@ -45,6 +45,7 @@
 
 [Sample](acceptance_test_spec.rb)
 
+- Use the most specific [selectors][] available.
```

Comment:
Does "most specific selector" mean favoring `button` selectors when using buttons? I can definitely see a case where some flexibility would be helpful - for example, with a selector like `link_or_button`, which would allow changing from a button to a link and vice-versa without a test failing.

Reply:
From a development perspective, being flexible enough to support finding or clicking on an `<a>` or `<button>` has appeal.

My goal with these guidelines is to encourage design and development exercises to center accessibility. With that in mind, the semantics of a `<a>` and `<button>` vary. The expectation for an `<a>` element are navigation focused, whereas `<button>` elements are action focused. To that end, assistive technologies like Screen Readers (e.g. VoiceOver) present Button and Links in different lists.

Being specific in your testing expectations forces you to _decide_ which element to use ahead of time. A failing test is valuable feedback that the element in question was changed, and provides you with an inflection point to re-consider or re-affirm that decision.

Reply:
That makes sense, thank you for explaining. Indeed there's a role difference between links and buttons and it really matters for accessibility.